processor/src/operations/horner_ops.rs

```diff
@@ -0,0 +1,502 @@
+    pub(super) fn op_horner_eval_base(&mut self) -> Result<(), ExecutionError> {
+        // --- read the values of the coefficients, over the base field, from the stack -----------
+        let coef = self.get_coeff_as_base_elements();
+
+        // --- read the evaluation point alpha from memory ----------------------------------------
+        let alpha = self.get_evaluation_point()?;
+
+        // --- read the accumulator values from stack ---------------------------------------------
+        let acc_old = self.get_accumulator();
+
+        // --- compute the updated accumulator values ---------------------------------------------
+        let acc_new =
+            coef.iter().rev().fold(acc_old, |acc, coef| QuadFelt::from(*coef) + alpha * acc);
+
+        // --- copy the stack ---------------------------------------------------------
+        self.stack.copy_state(0);
+
+        // --- update the accumulators ------------------------------------------------------------
+        self.stack.set(ACC_HIGH_INDEX, acc_new.to_base_elements()[1]);
+        self.stack.set(ACC_LOW_INDEX, acc_new.to_base_elements()[0]);
+
+        // --- set the helper registers -----------------------------------------------------------
+        self.populate_helper_registers(alpha);
+
+        Ok(())
+    }
+
+    /// Performs 4 steps of the Horner evaluation method on a polynomial with coefficients over
+    /// the quadratic extension field, i.e., it computes
+    ///
+    /// acc' = (((acc * alpha + c3) * alpha + c2) * alpha + c1) * alpha + c0
+    ///
+    ///
+    /// In other words, the intsruction computes the evaluation at alpha of the polynomial
+    ///
+    /// P(X) := c3 * X^3 + c2 * X^2 + c1 * X + c0
+    ///
+    /// As can be seen from the two equations defining acc', the instruction can be used in order
+    /// to compute the evaluation of polynomials of arbitrary degree by repeated invocations of
+    /// the same instruction interleaved with any operation that loads the next batch of 4
+    /// coefficients on the top of the operand stack, i.e., `mem_stream` or `adv_pipe`.
+    ///
+    /// The stack transition of the instruction can be visualized as follows:
+    ///
+    /// Input:
+    ///
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+---+----------+------+------+
+    /// | c0_1 | c0_0 | c1_1 | c1_0 | c2_1 | c2_0 | c3_1 | c3_0 | - | - | - | - | - |alpha_addr| acc1 | acc0 |
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+---+----------+------+------+
+    ///
+    ///
+    /// Output:
+    ///
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+---+----------+-------+-------+
+    /// | c0_1 | c0_0 | c1_1 | c1_0 | c2_1 | c2_0 | c3_1 | c3_0 | - | - | - | - | - |alpha_addr| acc1' | acc0' |
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+---+----------+-------+-------+
```

Comment:
Same question as above re the order of elements on the stack.
```

Reply:
Same reply here as in https://github.com/0xPolygonMiden/miden-vm/pull/1656#discussion_r1966968883.